### PR TITLE
docs: Update and combine troubleshooting guides

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR updates and combines the APM troubleshooting guides.

Screenshot 2023-05-16 at 7 30 52 PM

**Preview this PR:**

* [Troubleshoot](https://apm-server_10827.docs-preview.app.elstc.co/guide/en/apm/guide/master/troubleshoot-apm.html)
  * [Common problems](https://apm-server_10827.docs-preview.app.elstc.co/guide/en/apm/guide/master/common-problems.html)
  * [What happens when APM Server or Elasticsearch is down?](https://apm-server_10827.docs-preview.app.elstc.co/guide/en/apm/guide/master/server-es-down.html)
  * [APM Server response codes](https://apm-server_10827.docs-preview.app.elstc.co/guide/en/apm/guide/master/common-response-codes.html)
  * [Processing and performance](https://apm-server_10827.docs-preview.app.elstc.co/guide/en/apm/guide/master/processing-and-performance.html)
  * [APM Server binary debugging](https://apm-server_10827.docs-preview.app.elstc.co/guide/en/apm/guide/master/enable-apm-server-debugging.html)

### CODE REVIEW

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by simitt** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/common-problems.asciidoc [hunk 1]
```diff
@@ -1,99 +1,96 @@
...
-containing `[request]` in the APM Server logs.
-
-If no requests are logged, confirm that:
-
-. SSL isn't <<ssl-client-fails, misconfigured>>.
-. The host is correct. For example, if you're using Docker, ensure a bind to the right interface (for example, set
-`apm-server.host = 0.0.0.0:8200` to match any IP) and set the `SERVER_URL` setting in the {apm-agent} accordingly.
+[[data-indexed-no-apm-legacy]]
+[float]
+=== Data is indexed but doesn't appear in the APM UI
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/common-problems.asciidoc:23]: Understood. I'll update.

File: docs/common-problems.asciidoc [hunk 2]
```diff
@@ -1,99 +1,72 @@
...
+   "service.name": {
+      "full_name": "service.name",
+      "mapping": {
+         "name": {
+         "type": "text" <1>
+         }
+      }
+   }
+   }
+}
```
Comment (axw) [inline docs/common-problems.asciidoc:52]:
```suggestion
".ds-metrics-apm.transaction.1m-default-2023.04.12-000038": {
   "mappings": {
      "service.name": {
         "full_name": "service.name",
         "mapping": {
            "name": {
               "type": "text" <1>
            }
         }
      }
   }
}
```

File: docs/common-problems.asciidoc [hunk 3]
```diff
@@ -1,99 +1,72 @@
...
-APM agents provide auto-instrumentation for many popular frameworks and libraries.
-If the {apm-agent} is not auto-instrumenting something that you were expecting, data won't be sent to the {stack}.
-Reference the relevant {apm-agents-ref}/index.html[{apm-agent} documentation] for details on what is automatically instrumented.
+include::./tab-widgets/no-data-indexed-widget.asciidoc[]
 
+[[data-indexed-no-apm-legacy]]
 [float]
-[[common-response-codes]]
-=== APM Server response codes
+=== Data is indexed but doesn't appear in the APM UI
```
Comment (axw) [inline docs/common-problems.asciidoc:22]: Should "APM UI" be "APM app"? Not sure when we use one and when the other...

File: docs/common-problems.asciidoc
Comment (axw) [inline docs/common-problems.asciidoc:1]: The "SSL client fails to connect" section is pretty weird IMO. I realise it's preexisting, but I don't think I've ever read it before. Why use telnet to debug? How does one "Verify that the certificate is valid and that the hostname and IP match."?

I guess leave this as a followup, but perhaps we should just suggest that users run `curl` to check they can make a request to the server?

**Review by axw — APPROVED** [review]

File: docs/common-problems.asciidoc
Comment (bmorelli25) [reply docs/common-problems.asciidoc:1]: Thanks. I'll open an issue to address this.

**Pull request merged**